exercises/06-b/README.md

```diff
@@ -0,0 +1,317 @@
+    }
+
+    private static IHostBuilder CreateHostBuilder(string[] args)
+    {
+        var builder = Host.CreateDefaultBuilder(args);
+        builder.UseWindowsService();
+
+        builder.ConfigureLogging((ctx, logging) =>
+        {
+            // TODO: optionally choose a custom logging library
+            // https://docs.particular.net/nservicebus/logging/#custom-logging
+
+            logging.AddConfiguration(ctx.Configuration.GetSection("Logging"));
+
+            logging.AddEventLog();
+            logging.AddConsole();
+        });
+
+        builder.UseNServiceBus(ctx =>
+        {
+            var endpointConfiguration = new EndpointConfiguration("Billing");
+
+            var transport = endpointConfiguration.UseTransport<LearningTransport>();
+
+            // TODO: optionally choose a different serializer
+            // https://docs.particular.net/nservicebus/serialization/
+            endpointConfiguration.UseSerialization<NewtonsoftSerializer>();
+            endpointConfiguration.AddDeserializer<XmlSerializer>();
+
+            var conventions = endpointConfiguration.Conventions();
+            conventions.DefiningEventsAs(
+                type => (type.Namespace != null && type.Namespace.EndsWith(".Events"))
+            );
+
+            endpointConfiguration.AuditProcessedMessagesTo("audit");
+
+            endpointConfiguration.DefineCriticalErrorAction(OnCriticalError);
+
+            var displayName = System.Net.Dns.GetHostName();
+            var identifier = StringToGuid("Billing@" + displayName);
+
+            var endpointIdentity = endpointConfiguration.UniquelyIdentifyRunningInstance();
+            endpointIdentity.UsingCustomDisplayName(displayName);
+            endpointIdentity.UsingCustomIdentifier(identifier);
+
+            return endpointConfiguration;
+        });
+
+        return builder.ConfigureServices(services => { });
+
+    }
+
+    private static async Task OnCriticalError(ICriticalErrorContext context)
+    {
+        // https://docs.particular.net/nservicebus/hosting/critical-errors
+        var fatalMessage =
+            $"The following critical error was encountered:{Environment.NewLine}{context.Error}{Environment.NewLine}Process is shutting down. StackTrace: {Environment.NewLine}{context.Exception.StackTrace}";
+        EventLog.WriteEntry(".NET Runtime", fatalMessage, EventLogEntryType.Error);
+
+        try
+        {
+            await context.Stop().ConfigureAwait(false);
+        }
+        finally
+        {
+            Environment.FailFast(fatalMessage, context.Exception);
+        }
+    }
+}
+```
+
+### Step  3
+
+Add an assembly reference to `NServiceBus.Extensions.Hosting` and `Microsoft.Extensions.Hosting.WindowsServices`.
+
+### Step 4
+
+Assign the right value to `Console.Title`.
+
+### Step 5
+
+Add a critical error callback to method `OnCriticalError` to the `CreateConfiguration` function as documented on the following location:
```

Comment:
This is already done in the code above: https://github.com/Particular/Workshop.NServiceBus/pull/21/files#diff-a8a50ae453153e9758df23850ec4e726521c3b699fd60fdee4a47a45665f845dR90
Also, you reference documentation specific to Azure Host package?

Reply:
Good catch, but this was already here, as this is a copy from the original exercise.